### PR TITLE
Added files from the example relevant to file serving

### PR DESCRIPTION
**Pull request by mckennagalvin**
I added back the files that are used to serve files (like the MIME type resolution, request parsing for the requested URL, etc.)

I'm still a little confused by what the spec is asking -- it mentions having separate request handlers and having them implement the same interface. But right now we don't parse the details of the request until we get to `request_handler.cpp` -- at which point it would make more sense in my mind to just have two separate functions that deal with sending back the echo or file...

In any case, I created cpp and hpp files for `request_handler_echo` and `request_handler_static` in case we need to structure the code that way in the future (hoping the instructors can provide more clarification about this stuff...). Right now, `request_handler.cpp` is still in there for the echo requests and everything gets routed there, just so everything works as it did before.

I also refactored `main.cpp` a little to make parsing multiple options easier

There's also the issue of when to parse the given request, as all the echo request handler does is simply copy stuff from a buffer. In order to serve files, we'd need to use `request_parser` somewhere to extract which URL is being requested.

### CODE REVIEW

**Comment by mckennagalvin** [comment]
Update: got echo and file serving to work individually, now just need to figure out how to parse the request to determine which request handler to use in `connection.cpp`. I tried some stuff but left it commented out because I couldn't figure out how to log to the console for debugging purposes

File: src/connection.cpp [hunk 1]
```diff
@@ -43,20 +45,39 @@ void connection::stop() {
...
-    else {
+    else if (!result) {
       reply_ = reply::stock_reply(reply::bad_request);
       boost::asio::async_write(socket_, reply_.to_buffers(),
-          boost::bind(&connection::handle_write, shared_from_this(),
-            boost::asio::placeholders::error));
+        boost::bind(&connection::handle_write, shared_from_this(),
+          boost::asio::placeholders::error));
+    }
+    else {
```
Comment (andykuang1) [inline src/connection.cpp:73]: Is this necessary? I believe if (result) and if (!result) covers all situations. Never hurts to be safe though.

Comment (mckennagalvin) [reply src/connection.cpp:73]: I think result is a tribool, so it can have three states (true, false, indeterminate)

**Comment by andykuang1** [comment]
Seems like we have a duplicate of the html file we're trying to serve saved as static/Untitled Document~ (seems to be a temp file that got included unintentionally). Other than that everything looks fine, we'll work on getting it to actually serve files properly.

**Comment by mattdrag** [comment]
Looks like a lot of good foundation for us to use. Instead of doing the URI parsing in request_handler.cpp, where you put the TODO, I believe we can instead do that right in the handler. I will implement this in the next pull.

**Comment by mckennagalvin** [comment]
good catch on the static/Untitled Document~! That's definitely not supposed to be there

The reason for doing the URI parsing in `connection.cpp` was to determine which request_handler to send the request to (echo or static). If we were to move the parsing to request_handler we'd have to implement both echo and static file serving in the same file (which makes more sense to me, but it goes against what they said in the spec...)

**Pull request merged**